### PR TITLE
Enabled make check in travis.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubeflow/katib/issues/607

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

/cc @gaocegege @hougangliu @johnugeorge

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/608)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 language: go
 
 go:
-  - "1.10"
-  - "1.11"
-  - "1.12"
+  - "1.12.5"
```
Comment (gaocegege) [inline .travis.yml]: Can we use 1.12 here?

Comment (gyliu513) [reply .travis.yml]: I followed kubefed here https://github.com/kubernetes-sigs/kubefed/pull/902 , it is always good to use latest stable version.

Comment (gaocegege) [reply .travis.yml]: Personally, I do not think so. I think we should support multiple go versions.

Comment (gyliu513) [reply .travis.yml]: Different go version sometimes have different standard. In my previous PR, I found that `gofmt` always failed in go1.10 but succeed in go 1.12 and go 1.11, this caused the `make check` failed. Using the latest go can leverage the latest features in Go.

Comment (gaocegege) [reply .travis.yml]: Yeah, go 1.10 can be removed. Format check may be failed in 1.10. I also noticed it.

**Pull request merged**